File: coding_challenges_crickett/02_parser/json_parser.py
import json
import sys


def main():
    file_name = sys.argv[1]

    try:
        with open(file_name, 'r') as file:
            json.loads(file.read())
        print(f"{file_name} is valid JSON")
        exit(0)
    except json.JSONDecodeError:
        print(f"{file_name} is invalid JSON")
        exit(1)


if __name__ == "__main__":
    main()

File: coding_challenges_crickett/02_parser/json_parser.rb
# frozen_string_literal: true

require 'json'

if __FILE__ == $PROGRAM_NAME
  file_name = ARGV.first
  file = File.read(file_name)

  begin
    JSON.parse(file)
    puts "#{file_name} is valid JSON"
    exit(0)
  rescue JSON::ParserError
    puts "#{file_name} is invalid JSON"
    exit(1)
  end
end
